src/coreclr/jit/target.h

```diff
@@ -742,15 +739,15 @@ inline bool floatRegCanHoldType(regNumber reg, var_types type)
 
 extern const regMaskSmall regMasks[REG_COUNT];
 
-inline regMaskTP genRegMask(regNumber reg)
+inline SingleTypeRegSet genRegMask(regNumber reg)
```

Comment:
I am not sold on changing this from `regMaskTP` to `SingleTypeRegSet`. It is going to introduce ambiguity in the future about what register is being referred to.

As we discussed, I think we should isolate this handling to LSRA. LSRA can have its own context-dependent register sets where the first GP, float or predicate register all get mapped to 0, while the rest of the JIT can keep using the large register sets. LSRA can handle mapping these back to `regNumber` or `regMaskTP` (in the rare cases) to interact with the rest of the JIT.

For `instrDesc` we can stop storing `regMaskTP` in `instrDescCGCA` and just have an accessor method that maps the internal storage back to `regMaskTP`.